docs/rst/tutorials/core/deployment/ds_locators_runtime/ds_locators_runtime.rst

```diff
@@ -0,0 +1,369 @@
+to the file.
+
+Create a json file for each *client* and introduce the *server* IP address and port.
+Make sure the talker in the discovery environment 1 only contains its discovery server data:
+
+.. tabs::
+
+    .. tab:: Discovery Environment 0
+
+        .. literalinclude:: /resources/tutorials/core/deployment/ds_locators_runtime/environment_file_complete.json
+            :language: xml
+
+    .. tab:: Discovery Environment 1
+
+        .. literalinclude:: /resources/tutorials/core/deployment/ds_locators_runtime/environment_file.json
+            :language: xml
+
+The environment value can be either an absolute or relative path.
+Set up the environment file:
+
+.. code-block:: bash
+
+    export FASTDDS_ENVIRONMENT_FILE="environment_file.json"
+
+
+Run the example
+---------------
+
+After all the configuration has been set, run the discovery servers, and the talker and listener *clients*:
+
+.. tabs::
+
+    .. tab:: Discovery Environment 0
+
+        .. tabs::
+
+            .. tab:: Server_0
+
+                .. code-block:: bash
+
+                    fastdds discovery --server-id 0
+
+            .. tab:: Client_talker_0
+
+                .. code-block:: bash
+
+                    ros2 run demo_nodes_cpp talker
+
+            .. tab:: Client_listener
+
+                .. code-block:: bash
+
+                    ros2 run demo_nodes_cpp listener
+
+    .. tab:: Discovery Environment 1
+
+        .. tabs::
+
+            .. tab:: Server_1
+
+                .. code-block:: bash
+
+                    fastdds discovery --server-id 1
+
+            .. tab:: Client_talker_1
+
+                .. code-block:: bash
+
+                    ros2 run demo_nodes_cpp talker
+
+The expect output is that the listener would only receive the publications from the talker in the same discovery
+environment.
+Now, let's add the other *server* as a discovery server in the ``client_pub_1`` talker on run-time to let the listener
+receive its publications.
+
+Discovery Server on run-time
+----------------------------
+
+During execution, modify the listener ``json`` file to include the other *server* data, as follows:
+
+.. literalinclude:: /resources/tutorials/core/deployment/ds_locators_runtime/environment_file_complete.json
+    :language: xml
+
+After saving the file, the listener would discover the remain talker through the discovery server just set.
+
+.. uml::
+    :align: center
+
+    hide empty members
+
+    package "Discovery Environment 0" as de0{
+        (Server 0) as s0
+        (Client pub 0) as p0
+        (Client sub) as s
+        s <-[dashed]right- p0
+        s0 <-down- s
+        s0 <-down- p0
+        p0 -[hidden]left- s0
+    }
+
+    package "Discovery Environment 1" as de1{
+        (Server 1) as s1
+        (Client pub 1) as p1
+        p1 -up-> s1
+    }
+
+    de1 -[hidden]right- de0
+    s1 -[hidden]right- s0
+    p1 -[hidden]right- p0
+    p0 -[dotted]-> s1
```

Comment:
Again, we don't need this. I get that you're using the same file for both pub & sub. IDK, maybe we could considered separating them. What do you think? The thing is that looking at the figure, I fail to understand why would I do that for this particular demo